opam harsh upgrade from 1.1 to 1.2

hello,
please find some comments concerning my actual experience of opam upgrade.
Thanks for your tool by the way.

in :
http://opam.ocaml.org/doc/Install.html
it is said we can install OPAM in 2 minutes : ????
Maybe it is true on debian & co, but not from sources

install from sources :
I found there was a regression of the documentation compared to opam 1.1. There was a good start tutorial before, but now we are redirected to README.md

README.md :

after ./configure & make & install, reference to :
http://opam.ocaml.org/doc/Quick_Install.html
which do not work
I guess you meant  : http://opam.ocaml.org/doc/Usage.html

And the step about "opam init" is missing. There is no clear help on this in README.md. Maybe you should introduce a "setup" section in README.md and say something like :
- use "opam init" to setup basic opam in ~/.opam
- or use "opam init --help" in order to get more options

opam list [first command without initialising before]
[ERROR] No OPAM root found at /home/troit/.opam.
Please run 'opam init' first to initialize the state of OPAM.
=> you should tell we can use "opam init --help" also.
It is important because there is no more explanation on "opam init" in the help, neither in README.md nor in http://opam.ocaml.org/doc/Install.html

opam init --help
it is not easy to find that we can change ".opam" directory, and that it is possible with "--root".
Actually, I would like to install it for a group of people, and so install it on a shared directory

opam init --root /shared/directory/on/the/network
is VERY slow (more than 5 minutes), while duration seems normal on /my/local/directory
Usual access to /shared/directory/on/the/network is fast.

opam init --switch
I think in a previous doc it was told we could use this, but can not see it again.
how to know which switch can be used ? (not straitghtforward in the doc)
is it usefull ?

Best regards

--- a/README.md
+++ b/README.md
@@ -6,8 +6,8 @@ workflow.
 
 OPAM is created and maintained by [OCamlPro](http://www.ocamlpro.com).
 
-To get started, checkout the [Quick
-Install](http://opam.ocaml.org/doc/Quick_Install.html) guide.
+To get started, checkout the [Install](http://opam.ocaml.org/doc/Install.html)
+and [Usage](http://opam.ocaml.org/doc/Usage.html) guides.
 
 ## Compiling this repo
 

--- a/doc/pages/Install.md
+++ b/doc/pages/Install.md
@@ -148,8 +148,8 @@ You can also download the full archives, including OPAM dependencies:
 * [1.1.0](http://www.ocamlpro.com/pub/opam-full-1.1.0.tar.gz)
   MD5 (opam-full-1.1.0.tar.gz) = d6e2f56b10c0be73b5677963e6659d24
 
-Follow the instructions in `README.md` to get OPAM built and installed from
-there.
+Follow the instructions in [`README.md`](https://github.com/ocaml/opam#readme)
+to get OPAM built and installed from there.
 
 
 #### Using ocamlbrew

--- a/doc/pages/Usage.md
+++ b/doc/pages/Usage.md
@@ -19,6 +19,11 @@ details.
 ## Basics
 
 ```
+# ** Get started **
+opam init            # Initialise ~/.opam using an already installed OCaml
+opam init --compiler 4.02.1
+                     # Initialise with a freshly compiled OCaml 4.02.1
+
 # ** Lookup **
 opam list -a         # List all available packages
 opam search QUERY    # List packages with QUERY in their name or description
@@ -47,7 +52,8 @@ OCaml version -- `opam install PACKAGE` will give you the reason.
 ### opam init
 
 OPAM needs to initialise its internal state in a `~/.opam` directory to work.
-This command can also take care of installing a version of OCaml if needed.
+This command can also take care of installing a version of OCaml there if
+needed, using the `--comp VERSION` option.
 
 To operate as expected, some variables need to be set in your environment. You
 will be prompted to update your configuration, and given instructions on how

--- a/src/client/opamArg.ml
+++ b/src/client/opamArg.ml
@@ -680,10 +680,11 @@ let init =
   let doc = init_doc in
   let man = [
     `S "DESCRIPTION";
-    `P "The $(b,init) command creates a fresh client state.  This initializes OPAM \
-        configuration in $(i,~/.opam) and configures a default package repository.";
+    `P "The $(b,init) command creates a fresh client state. This initializes OPAM \
+        configuration in $(i,~/.opam) (or the given $(b,--root)) and configures \
+        the initial remote package repository.";
     `P "Once the fresh client has been created, OPAM will ask the user if he wants \
-        $(i,~/.profile) (or $i,~/.zshrc, depending on his shell) and $(i,~/.ocamlinit) \
+        $(i,~/.profile) (or $(i,~/.zshrc), depending on his shell) and $(i,~/.ocamlinit) \
         to be updated. \
         If $(b,--auto-setup) is used, OPAM will modify the configuration files automatically, \
         without asking the user. If $(b,--no-setup) is used, OPAM will *NOT* modify \
@@ -694,7 +695,7 @@ let init =
   ] in
   let jobs = mk_opt ["j";"jobs"] "JOBS" "Number of jobs to use when building packages." Arg.int OpamGlobals.default_jobs in
   let compiler =
-    mk_opt ["comp"] "VERSION" "Which compiler version to use." compiler OpamCompiler.system in
+    mk_opt ["compiler"] "VERSION" "Which compiler version to use." compiler OpamCompiler.system in
   let repo_name =
     let doc = Arg.info ~docv:"NAME" ~doc:"Name of the repository." [] in
     Arg.(value & pos ~rev:true 1 repository_name OpamRepositoryName.default & doc) in

--- a/src/client/opamState.ml
+++ b/src/client/opamState.ml
@@ -2628,7 +2628,9 @@ let check f =
   || not (OpamFilename.exists (OpamPath.config root)) then
     OpamGlobals.error_and_exit
       "No OPAM root found at %s.\n\
-       Please run 'opam init' first to initialize the state of OPAM."
+       Please run 'opam init' to initialize the state of OPAM, or \
+       specify '--root'.\n\
+       See 'opam init --help' for details."
       (OpamFilename.Dir.to_string root);
 
   match f with

